### PR TITLE
Enable unnecessary code rules

### PR DESCRIPTION
**Pull request by stan-sz**
Closes: #798

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -44,4 +44,21 @@ csharp_new_line_before_else = true
...
 csharp_new_line_before_finally = true
 csharp_new_line_before_members_in_object_initializers = true
-csharp_new_line_before_members_in_anonymous_types = true
\ No newline at end of file
+csharp_new_line_before_members_in_anonymous_types = true
+
+# https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/unnecessary-code-rules
+
+# Avoid unused private fields
+dotnet_diagnostic.CA1823.severity = error
```
Comment (WeihanLi) [inline .editorconfig:52]: Maybe better add a document link
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/

**Comment by brendandburns** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**